### DISCUSSION TITLE
RESTEasy resource classes identification

### DISCUSSION THREAD

**Discussion by gquintana** [quarkusio/quarkus#34470] [Q&A] [answered]
The JAX RS 3.1 Spec says

> A resource class is a Java class that uses JAX-RS annotations to implement a corresponding Web resource. Resource classes are POJOs that have at least one method annotated with @Path or a request method designator.

https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1.html#resource-classes

As far as I understand, this class should be detected as a JAX RS resource:
```
public class SimpleResource {
  @GET
  @Path("hello")
  @Produces(MediaType.APPLICATION_JSON)
  public String sayHello() {
    return "Hello world!";
  }

  @GET
  @Path("bye")
  @Produces(MediaType.APPLICATION_JSON)
  public String sayBye() {
    return "Bye world!";
  }
}
````
But it is not.
I have to add a `@Path` annotation on the class
```
@Path("")
public class SimpleResource {
```
Is it an intended behaviour?

**Comment by gsmet** [comment]
Hey @gquintana ,

Is it with RESTEasy (Classic) or RESTEasy Reactive?

**Comment by gquintana** [comment]
@gsmet resteasy-reactive 3.1.0.Final

**Reply by gsmet** [reply]
@geoand rings a bell?

**Comment by geoand** [comment, marked as answer]
Yes, this is expected behavior
